awslabs/gluon-ts
# Update `gluonts.dataset.split` code, test, docs

*Description of changes:* Follow up to #2031, #2157 with further adjustments:
- Rename `TestIterable` into `TestDataset`
- Add `input` and `label` properties to `TestDataset`
- Rename `split_date` and `split_offset`, in `DateSplitter` and `OffsetSplitter`, to `date` and `offset` respectively
- Update the module docstring and make it a (running) doctest
- Remove subpackage in favor of the module directly
- Add/fix type annotations where needed
- Change tests and documentation notebooks accordingly


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.


**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup
